temporal-sdk/src/main/java/io/temporal/internal/worker/EagerActivitySlotsReservation.java

```diff
@@ -66,34 +74,44 @@ public void applyToRequest(RespondWorkflowTaskCompletedRequest.Builder mutableRe
   public void handleResponse(RespondWorkflowTaskCompletedResponse serverResponse) {
     int activityTasksCount = serverResponse.getActivityTasksCount();
     Preconditions.checkArgument(
-        activityTasksCount <= this.outstandingReservationSlotsCount,
+        // TODO: This check seems backwards?
```

Comment:
Why is it backwards?

Reply:
Ugh, nevermind, I'm used to the check being "if this condition is met, it's a problem" but I spaced and forgot `Preconditions` is "if this check _isn't_ met, it's a problem".